app/queries/subscription_content_change_query.rb

```diff
@@ -0,0 +1,53 @@
+class SubscriptionContentChangeQuery
+  def initialize(subscriber:, digest_run:)
+    @subscriber = subscriber
+    @digest_run = digest_run
+  end
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def call
+    present
+  end
+
+private
+
+  attr_reader :subscriber, :digest_run
+
+  Result = Struct.new(:subscription_id, :subscription_uuid, :subscriber_list_title, :content_changes)
+
+  def present
+    grouped_content_changes.map do |(subscription_id, subscription_uuid, subscriber_list_title), content_changes|
+      Result.new(
+        subscription_id,
+        subscription_uuid,
+        subscriber_list_title,
+        content_changes.map { |attrs| hydrate_content_change(attrs) },
+      )
+    end
+  end
+
+  def hydrate_content_change(attrs)
+    ContentChange.new(
+      attrs.except("subscription_id", "subscription_uuid", "subscriber_list_title")
+    )
```

Comment:
This seems a bit unorthadox - I can see why you've done it though. I think there might be a way around it though. I'll just have a quick tinker.